StantDrawing/BrushTool/BrushToolView.swift

```diff
@@ -187,27 +221,115 @@ public class BrushToolView: UIView {
-        configureGroupLayout(buttons: groupPencilSizeButtons, groupView: groupPencilSizeView, groupViewOirigin: origin, addAsSubview: false)
-        groupPencilSizeView.isHidden = true
+        // Pencil Icon
+        let pencilIconView   = UIView()
         
-        setAllGroupdPencilSizeButtonsAsNotClicked()
+        let originIconView = CGPoint(x: 0, y: 0)
+        let sizeIconView   = CGSize(width: groupViewWidth, height: 60)
+        let frameIconView  = CGRect.init(origin: originIconView, size: sizeIconView)
+        
+        pencilIconView.frame              = frameIconView
+        pencilIconView.backgroundColor    = UIColor(hex: "F0F0F0")
+        pencilIconView.layer.cornerRadius = pencilIconView.frame.width / 2
+        
+        let originIconImg = CGPoint(x: (groupViewWidth - 25) / 2, y: 18)
+        let sizeIconImg   = CGSize(width: 25, height: 25)
+        let frameIconImg  = CGRect.init(origin: originIconImg, size: sizeIconImg)
+        
+        let pencilIconImageView = UIImageView(image: UIImage(named: "pencil", in: bundle, compatibleWith: nil))
+        
+        pencilIconImageView.accessibilityIdentifier = "pencilSizeReferenceIdentifier"
+        pencilIconImageView.frame                   = frameIconImg
+        
+        pencilIconView.addSubview(pencilIconImageView)
 
-        if let uiButton = groupPencilSizeButtons.first?.uiButton {
-            setBtnSizeAsClicked(button: uiButton)
-        }
+        // Separator
+        let separatorView   = UIView()
+        let originSeparator = CGPoint(x: 10, y: pencilIconView.frame.origin.y + pencilIconView.frame.height + 10)
+        let sizeSeparator   = CGSize(width: groupViewWidth - 20, height: 1)
+        let frameSeparator  = CGRect.init(origin: originSeparator, size: sizeSeparator)
+
+        separatorView.frame = frameSeparator
+        separatorView.backgroundColor = UIColor.black
+        separatorView.alpha = 0.2
+
+        // Group Pencil Buttons
+        let groupPencilBtns = UIView()
+        let origin          = CGPoint(x: 0, y: separatorView.frame.origin.y + separatorView.frame.height + 10)
+        
+        groupPencilBtns.frame.origin = origin
+        
+        configureGroupLayout(buttons: groupPencilSizeButtons, groupView: groupPencilBtns, groupViewOirigin: origin, addAsSubview: false)
+        
+        setAllGroupdPencilSizeButtonsAsNotClicked()
+        
+        setBtnSizeAsClicked(button: groupPencilSizeButtons[1].uiButton)
+        
+        // GroupPencilSizeView Configs
+        let height    = pencilIconView.frame.size.height + separatorView.frame.size.height + groupPencilBtns.frame.size.height
+        let sizeGroup = CGSize(width: self.frame.width, height: height)
+        groupPencilSizeView.frame.size = sizeGroup
+        groupPencilSizeView.isHidden   = true
+        
+        groupPencilSizeView.addSubview(pencilIconView)
+        groupPencilSizeView.addSubview(separatorView)
```

Comment:
I think this method is so big that need some comments. Try to extract every function to a specific method, like groupPencilSizeViewConfiguration() that groups from line 269 to 276.